### PR TITLE
FcsErrorsStageProcessBase: workaround against leaking fcs types

### PR DESCRIPTION
**Pull request by DedSec256**

### CODE REVIEW

File: ReSharper.FSharp/src/FSharp/FSharp.Psi.Services/src/Daemon/Highlightings/FcsErrors.xml [hunk 1]
```diff
@@ -406,9 +406,7 @@
   </Error>
 
   <Error staticGroup="FSharpErrors" name="TypeEquation" ID="FS0001: Type equation error">
-    <Parameter type="FSharpType" name="expectedType" isValid="skip"/>
```
Comment (DedSec256) [inline ReSharper.FSharp/src/FSharp/FSharp.Psi.Services/src/Daemon/Highlightings/FcsErrors.xml:409]: Removed now, as it is not used anywhere

File: ReSharper.FSharp/src/FSharp/FSharp.Psi.Services/src/Daemon/Highlightings/FSharpErrorUtil.fs [hunk 2]
```diff
@@ -11,6 +12,12 @@ open JetBrains.ReSharper.Plugins.FSharp.Util
 open JetBrains.ReSharper.Psi.ExtensionsAPI
 open JetBrains.ReSharper.Psi.Tree
 
+type DiagnosticTypeInfo =
+    { IsTupleType: bool; TypeString: string } with
+    static member Create(fcsType: FSharpType, displayContext: FSharpDisplayContext) =
```
Comment (auduchinok) [inline ReSharper.FSharp/src/FSharp/FSharp.Psi.Services/src/Daemon/Highlightings/FSharpErrorUtil.fs:17]: Please add an empty line after the type representation.

File: ReSharper.FSharp/src/FSharp/FSharp.Psi.Services/src/Daemon/Highlightings/FSharpErrorUtil.fs [hunk 3]
```diff
@@ -11,6 +12,12 @@ open JetBrains.ReSharper.Plugins.FSharp.Util
 open JetBrains.ReSharper.Psi.ExtensionsAPI
 open JetBrains.ReSharper.Psi.Tree
 
+type DiagnosticTypeInfo =
```
Comment (auduchinok) [inline ReSharper.FSharp/src/FSharp/FSharp.Psi.Services/src/Daemon/Highlightings/FSharpErrorUtil.fs:15]: Let's call it something like `FSharpDiagnosticTypeInfo`, so it's clearer it's coming from the plugin?

**Review by auduchinok — COMMENTED** [review]
Cool, thanks! Just a small stylistic suggestion from me.

**Review by auduchinok — APPROVED** [review]

**Pull request merged**